nameko/containers.py

```diff
@@ -171,8 +170,8 @@ def __init__(self, service_cls, config, worker_ctx_cls=None):
         self.started = False
         self._worker_pool = GreenPool(size=self.max_workers)
 
-        self._active_threads = {}
-        self._protected_threads = set()
+        self._worker_threads = WeakKeyDictionary()
```

Comment:
given that they weren't before, and we use `link` pretty well, i think i'd prefer to avoid weakrefs